gui/src/WorkspaceView/WorkspaceViewMainWindow.tsx

```diff
@@ -81,8 +82,7 @@ const WorkspaceViewMainWindow: FunctionComponent<Props> = ({data, selection, sel
     const scalarDataRange: [number, number] = useMemo(() => ([vMin, vMax]), [vMin, vMax])
 
     const {svMin, svMax} = useMemo(() => {
-        const names = Object.values(selection.selectedSurfaceScalarFieldNames).filter(a => (a !== undefined)).map(a => (a as string))
-        const scalarFields = names.map(name => (data.surfaceScalarFields.filter(x => (x.name === name))[0]))
+        const scalarFields = data.surfaceScalarFields.filter(x => (selection.selectedSurfaceScalarFieldNames[x.surfaceName]?.includes(x.name)))
```

Comment:
So to make sure I'm understanding this change: previously, when calibrating the range for values in a scalar field, we would pull a list of every surface scalar field name that was selected, and then pull the scalar fields corresponding to those names.

After the change, we pull the scalar fields satisfying the criteria: the selected-field-name of the surface this field belongs to, matches this field's name.

So the difference is in the first case, we're finding the range by looking at every field whose name matches a selection--whether it's selected on its corresponding surface or not--while after the change, we are only pulling those scalar fields which are selected on their surfaces.

Is that right?

Reply:
Sort of but not quite. There was actually a bug in the old code. It was only finding the scalar fields with the given names and was not requiring the surfaceNames to match. So it was picking off the first field with the corresponding name, rather than the unique field with the corresponding name and surfaceName

Reply:
Ah, okay, gotcha.